adminapi/cli.py

```diff
@@ -0,0 +1,131 @@
+        action='store_true',
+        help='Make sure exactly one server matches with the query',
+    )
+    parser.add_argument(
+        '-a',
+        '--attrs',
+        nargs='+',
+        default=['hostname'],
+        help='The attributes to fetch',
+    )
+    parser.add_argument(
+        '-o',
+        '--orderby',
+        nargs='+',
+        default=['hostname'],
+        help='Attributes to order by the result',
+    )
+    parser.add_argument(
+        '-r',
+        '--reset',
+        nargs='+',
+        default=[],
+        help='Reset multi attributes',
+    )
+    parser.add_argument(
+        '-u',
+        '--update',
+        nargs='+',
+        type=attr_value,
+        default=[],
+        help='The attributes to update',
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    servers = (
+        QuerySet(
+            # TODO: Avoid .join()
+            filters=parse_query(' '.join(args.quals)),
+            auth_token=_api_settings['auth_token'],
+            timeout=_api_settings['timeout_dataset'],
+        )
+        .restrict(*(args.attrs + args.reset + [k for k, v in args.update]))
+        .order_by(*args.orderby)
+    )
+
+    if args.one:
+        if not servers:
+            raise Exception('No servers')
+        if len(servers) > 1:
+            raise Exception('Multiple servers')
+
+    changes = bool(args.reset or args.update)
+    for server in servers:
+        print(server)
+        if changes:
+            apply_changes(server, args.reset, args.update)
+        print_server(server, args.attrs)
+    if changes:
+        servers.commit()
+
+
+def attr_value(arg):
+    arg_split = tuple(arg.split('='))
+    if len(arg_split) != 2:
+        raise ArgumentTypeError('You need to pass an attribute=value')
+    return arg_split
+
+
+def apply_changes(server, reset, update):
+    for attribute_id in reset:
+        server[attribute_id].clear()
+    for attribute_id, value in update:
+        if type(server[attribute_id]) is MultiAttr:
+            server[attribute_id].add(value)
+        elif type(server[attribute_id]) is bool:
+            server[attribute_id] = strtobool(value)
+        elif type(server[attribute_id]) is int:
+            server[attribute_id] = int(value)
+        else:
+            server[attribute_id] = value
+
+
+def print_server(server, attribute_ids):
+    values = []
+    for attribute_id in attribute_ids:
+        if attribute_id not in server:
+            values.append('{N/A}')
+            continue
+
+        value = server[attribute_id]
+
+        # Temporary hack
+        if attribute_id == 'hostname':
+            if not any(value.endswith(d) for d in [
+                '.ig.local',
+                '.innogames.net',
+            ]):
+                value += '.ig.local'
+
+        if value in [None, True, False]:
+            value = '{{{}}}'.format(str(value).lower())
+
+        values.append(value)
+
+    print(*values, sep='\t')
```

Comment:
It seems you just print the values. Wouldn't it be helpful to also print the related attribute names?

Reply:
Not by default.  The user already knows what attributes they required.  Also, I am trying to keep this compatible with the existing executable.  Typical usage is something like `gsh $(igserver project=foe function=db)`.

Reply:
Fair enough then.